### PR TITLE
Advanced Throwing - Cleanup

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.
- Requires #10492 to avoid merge conflicts.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/advanced_throwing/functions/fnc_getMuzzle.sqf
Comment (johnb432) [inline addons/advanced_throwing/functions/fnc_getMuzzle.sqf:1]: This function is only used once and we can get the muzzle from `currentThrowable`.
Does anyone know why this function exists?

Comment (PabstMirror) [reply addons/advanced_throwing/functions/fnc_getMuzzle.sqf:1]: I don't know, maybe we thought there was a use in getting the muzzle for a grenade not selected
but I don't see any use for it now.

**Comment by PabstMirror** [comment]
noticed some weirdness with selecting grenades by hitting shift-g multiple times
but it seems to be fixed by [#9488 (see note)](https://github.com/acemod/ACE3/pull/9488#issuecomment-2492324380) and is probably unrelated to this PR

**Review by PabstMirror — APPROVED** [review]

File: addons/advanced_throwing/functions/fnc_prime.sqf [hunk 1]
```diff
@@ -44,11 +44,11 @@ if (_ammoCount == 1) then {
 _unit setAmmo [_muzzle, 0];
 
 private _throwableType = getText (_config >> "ammo");
+private _ammoConfig = configFile >> "CfgAmmo" >> _throwableType;
 
 // Handle weird scripted grenades (RHS) which could cause unexpected behaviour
-private _nonInheritedCfg = configProperties [configFile >> "CfgAmmo" >> _throwableType, 'configName _x == QGVAR(replaceWith)', false];
-if ((count _nonInheritedCfg) == 1) then {
-    _throwableType = getText (_nonInheritedCfg select 0);
+if (inheritsFrom (_ammoConfig >> QGVAR(replaceWith)) isEqualTo _ammoConfig) then {
```
Comment (johnb432) [inline addons/advanced_throwing/functions/fnc_prime.sqf:50]: Fyi `QGVAR(replaceWith)` isn't actually used anywhere, but I'm keeping it for BWC, despite it not being documented.

**Pull request merged**